File: SourceCodeExample/partition.py
def partition(work, begin, end, pivot):

    value = work[pivot]
    work[pivot] = work[begin]

    i = begin + 1
    j = end - 1
for 
    while i < j:
        while i < j and work[j] > value:
            j -= 1
        while i < j and work[i] < value:
            i += 1

        if i < j:
            tmp = work[i]
            work[i + 1] = work[j]
            work[j - 1] = tmp

    if i >= end or work[i] > value:
        i -= 1

    work[begin] = work[i]
    work[i] = value

    return i

File: SourceCodeExample/partition.cpp
#include <iostream>

int partition(double work[  ], int begin, int end, int pivot) {
    double value = work[pivot];
    work[pivot] = work[begin];
    int i = begin + 1;
    int j = end - 1;
    while (i < j) {
        while (i < j && work[j] > value) {
            --j;
        }
        while (i < j && work[i] < value) {
            ++i;
        }
        if (i < j) {
            double tmp = work[i];
            work[i++] = work[j];
            work[j--] = tmp;
        }
    }
    if (i >= end || work[i] > value) {
        --i;
    }
    work[begin] = work[i];
    work[i] = value;
    return i;
}

